Shell/Property.cpp

```diff
@@ -1059,6 +1063,139 @@ bool Property::hasXEqualsY(const Formula* f)
   return false;
 } // Property::hasXEqualsY(const Formula* f)
 
+
+/**
+ * True if f should clausify to only introduce constants as Skolems.
+ *
+ * @warning Works correctly only closed formulas!
+ */
+bool Property::onlyExistsForallPrefix(UnitList* units)
+{
+  Stack<const Formula*> forms;
```

Comment:
If these always move in triplets, it would make the function a bit more robust in my opinion to use a stack of 3-tuples (or a local `struct`).

Reply:
Looks nice, thank you!